src/app/services/theme/theme.service.ts

```diff
@@ -0,0 +1,202 @@
+import { DOCUMENT } from "@angular/common";
+import { Inject, Injectable } from "@angular/core";
+import { IS_SERVER_PLATFORM } from "src/app/app.helper";
+
+export const bawThemes = [
+  "highlight",
+  "primary",
+  "secondary",
+  "success",
+  "info",
+  "warning",
+  "danger",
+  "light",
+  "dark",
+] as const;
+
+export const bawThemeVariants = [
+  "",
+  "-lighter",
+  "-lightest",
+  "-darker",
+  "-darkest",
+] as const;
+
+export type BawTheme = typeof bawThemes[number];
+export type BawThemeVariant = typeof bawThemeVariants[number];
```

Comment:
Publically, I don't think these colour variables need to be exposed as component strings.

I.e. we should never see --darker by itself.

Reply:
I'm unsure of what solution you are looking for. Often I want to use the palette by itself. I also often use it in combination with the variant (whilst not knowing which theme is being applied). Ie. I will apply `-darker` to a theme, without knowing which theme is in use.

Reply:
That's an interesting use case. 

In which case:  a lookup function is more useful?

`darker("primary") => "--primary-darker"`

Reply:
A viable solution, however it assumes that we a) will only have `base`, `lighter`, `lightest`, etc (bootstrap may move to a 100, 200, 300, ..., 900 solution in the future), and b) will never want to generate an array of all palettes and variants (ie. the theme-template page), as I would have to make this array anyway. It seems like a more cumbersome solution

Reply:
Ok the distinct issue I have is we should not be exposing fragments of variables names from a public API.

If you want to expose groups of colours, then make a structure that reflects that.

If you want to expose variants of a colour, then make a helper that does that.